catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/AccessControlAccessPlugin.java

```diff
@@ -76,33 +104,27 @@ public CreateRequest processPreCreate(CreateRequest input) {
   public UpdateRequest processPreUpdate(
       UpdateRequest input, Map<String, Metacard> existingMetacards) throws StopProcessingException {
 
-    Function<Metacard, Metacard> oldVersionOfMetacard =
-        (update) -> {
-          Optional<Metacard> oldMetacard =
-              Optional.ofNullable(existingMetacards.get(update.getId()));
-          if (oldMetacard.isPresent()) {
-            return oldMetacard.get();
-          }
-          return null;
-        };
-
-    boolean foundInaccessibleMetacard =
-        input
-            .getUpdates()
-            .stream()
-            .map(Map.Entry::getValue)
-            .filter(
-                newVersionOfMetacard ->
-                    isAccessControlUpdated(
-                        oldVersionOfMetacard.apply(newVersionOfMetacard), newVersionOfMetacard))
-            .filter(Objects::nonNull)
-            .filter(mc -> !subjectIsAccessAdmin.test(mc) && !subjectIsOwner.test(mc))
-            .findFirst()
-            .isPresent();
-
-    if (foundInaccessibleMetacard) {
-      throw new StopProcessingException(
-          "Cannot update metacard(s). Subject cannot change access control permissions because they are not in the assigned access-administrators list.");
+    Function<Metacard, Metacard> oldMetacard =
+        update -> Optional.ofNullable(existingMetacards.get(update.getId())).orElse(null);
```

Comment:
:question: Am I missing something or the same result can be achieved with `update -> existingMetacards.get(update.getId());`?

Reply:
Nice catch. I'm tweaking this slightly so we can protect against passing a `null` key to `Map.get`.

Reply:
Could you please point out where this prevention takes place, I am still not getting it?

Reply:
https://github.com/codice/ddf/pull/4233/files#diff-97b1a9614ec9df5e96f1f39c442a3baeR124